Adds ravens

Raven567 opened the pull request:
![Ravendead](https://user-images.githubusercontent.com/28615254/195206488-5f3a693b-d720-41d9-934b-29b8abf8233c.png)

![Ravenleft4](https://user-images.githubusercontent.com/28615254/195603242-9837a778-1292-4f7c-ad38-7532fc09a276.png)

![Ravenright4](https://user-images.githubusercontent.com/28615254/195603298-a09c96e1-c8dd-4836-b738-8546e118e697.png)

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do
<!--  -->
Adds a pet raven, updates the animal.dmi, with raven sprites ,and animations. Also adds the raven.dm script
## Why It's Good For The Game
<!--  -->
Because ravens are cool! And we need ravens!
## Images of changes
<!--
 
 -->
![image](https://user-images.githubusercontent.com/28615254/196016948-06eaee7e-92ee-4508-aff2-b1b168bdb967.png)
![image](https://user-images.githubusercontent.com/28615254/196016998-bcf0e35d-fa66-4439-94f3-c587207fbdc9.png)


added 47 sprites It is roughly similar number as parrot. 
## Testing
<!-- -->
 I compiled the game, and used admin "animalize" in the player_pannel, to make sure all the animations, and interactions are working correctly.
![image](https://user-images.githubusercontent.com/28615254/195436507-cd85de8f-9e38-407e-885c-109ffcfdf036.png)
I recently did a large update to the branch; adds a raven pet, "Nevermore to each of the stations chapels.
![image](https://user-images.githubusercontent.com/28615254/195601338-1dccdb46-152e-450d-9290-7963209bcb60.png)


## Changelog
:cl:
add: adds Raven animals! And the pet nevermore!

/:cl:

This is my first commit ever, I expect some criticism, and possible revision?
![Ravendead](https://user-images.githubusercontent.com/28615254/195206488-5f3a693b-d720-41d9-934b-29b8abf8233c.png)
![Ravenleftflyt](https://user-images.githubusercontent.com/28615254/196016870-7e75e285-292d-4fda-ae87-4099d79d2c89.gif)

![Ravenleft4](https://user-images.githubusercontent.com/28615254/195603242-9837a778-1292-4f7c-ad38-7532fc09a276.png)


![RavenflyRear-export](https://user-images.githubusercontent.com/28615254/196016867-0451ef12-788a-455d-85e3-dab5528fe93e.gif)

![Ravenflyfront](https://user-images.githubusercontent.com/28615254/196016865-3aac6542-e266-4faa-b301-578a202cc65b.gif)

![RavenRightflyt](https://user-images.githubusercontent.com/28615254/196016858-3344b49d-bb9d-4cd9-b873-0bfb4ae14616.gif)

FloFluoro commented on code/modules/mob/living/simple_animal/friendly/raven.dm:
```diff
@@ -0,0 +1,26 @@
+/mob/living/simple_animal/pet/raven
+	name = "raven"
+	desc = "Qouth the raven never more! Just now in space!."
```
```suggestion
	desc = "Quoth the raven nevermore! Just now in space!"
```

Some spelling and punctuation fixes

McRamon commented:
Please include easily seen icons on your pr

Miraviel commented:
Fuck yeah, birds!

They look huge, though, compared to the other animals. We could have giant ravens, but we might want to call them "giant ravens", just to make sure it is intentionally this big!

Edan52 commented on code/modules/mob/living/simple_animal/friendly/raven.dm:
```diff
@@ -0,0 +1,26 @@
+	icon_state = "raven"
+	icon_living = "raven"
+	icon_dead = "raven_dead"
+	speak_emote = list("caws","caw")
```
Caw doesn't make much sense here

Raven567 replied:
Fixed, changed to "caws"

Edan52 commented on code/modules/mob/living/simple_animal/friendly/raven.dm:
```diff
@@ -0,0 +1,26 @@
+	speak_emote = list("caws","caw")
+	health = 7
+	maxHealth = 7
+	attacktext = "pecks violently"
```
This also doesn't make much sense in a sentence to me.

Raven567 replied:
Give me a suggestion maybe? I think pecks is a pretty understandable attack reaction for a bird? I changed it to bites in response.

Edan52 replied:
pecks by itself is fine, just that it'll say `raven pecks violently [name]` if it were like that, which doesn't make much sense to me.

Raven567 replied:
Oh I get it.

Sirryan2002 reviewed (changes requested):
Sprite needs some work. Please put the sprite in the PR description.

- The raven needs to be smaller, more on par with bunnies or parrots
- There needs to be more shading on the raven

Raven567 commented:
![image](https://user-images.githubusercontent.com/28615254/195476818-9f4ed3d5-4696-4f7a-ab4e-6723bd914772.png)

Raven567 commented:
> Code is looking good, go ahead and get your forum account linked.

![image](https://user-images.githubusercontent.com/28615254/195476851-379b89ef-28b1-4b2f-9a3b-6862bae5c101.png)

Raven567 commented:
I still don't really know how things are done around here. So I am writing this manifesto. as part for the development team to review, and community as well. 

I feel adding a new mob, and increasing variety to the game would be a positive outcome. However my considerations are as follows.
-I wish to add flying at the later date, after this push request. I made this really to test the waters, and I want too keep things simple, however I have already started work on the flight animations, and I believe its easily achievable. 
-I would like it to be a playable pet somewhere, I have read through past commits and found that poly caused chaos, and was abused by players using it to rat out traitors and such. I think this is to be avoided and I am seeking council on where the raven should be put, and what abilities it should have. 
-Pixel art, I have received tremendous help form the community, and any suggestions on this regard will be taken to heart most likely. I want the best looking raven possible.
And thank you all very much for your time, and reading my nonsense!

Sirryan2002 commented on code/modules/mob/living/simple_animal/friendly/raven.dm:
```diff
@@ -0,0 +1,26 @@
+	response_help  = "pets"
+	response_disarm = "shoos"
+	response_harm   = "kicks"
+	ventcrawler = 2
```
```suggestion
	ventcrawler = VENTCRAWLER_ALWAYS
```
we have a define for this value

Raven567 replied:
Massive props to you sir.

Sirryan2002 reviewed (commented):
few small things

hal9000PR commented:
accounts linked

Sirryan2002 reviewed (approved):
code is fine

McRamon commented:
Dead sprite is really something to be improved on. Or at least made similar to Poly

Raven567 commented:
> Dead sprite is really something to be improved. Or at least made similar to Poly

I need to do a refactor, people really want flying and perching and I need probably the weekend to work on this.

Raven567 commented:
Poly is too much to refactor in one night!!! Ill work on the bird refactor another time! 
However I have added functionality to my raven, it can now caw! I also added functionality for flying when the pixel art is done!

AffectedArc07 commented on code/modules/mob/living/simple_animal/friendly/raven.dm:
```diff
@@ -0,0 +1,57 @@
+var/cooldown = world.time//DO MOBS NOT HAVE A COOLDOWN VARIBLE!
+/mob/living/simple_animal/pet/raven
+	name = "raven"
```
This var needs scoping to the mob level, and possibly naming something more sane.

Raven567 replied:
Alright ill throw it into the mob level. I am not sure what to name it, I am probably not qualified enough for sanity yet.

Denthamos commented:
Is there a flying animation or does it just walk on the ground?

Sirryan2002 reviewed (changes requested):
rescinding code approval due to new changes

Sirryan2002 commented on code/modules/mob/living/simple_animal/friendly/raven.dm:
```diff
@@ -0,0 +1,57 @@
+	set category = "Raven"
+	set desc = "Sit on a nice comfy perch."
+
+	if(icon_state== "raven")
```
```suggestion
	if(icon_state == "raven")
```
this shouldn't rely on checking the icon_state for the logic

Sirryan2002 commented on code/modules/mob/living/simple_animal/friendly/raven.dm:
```diff
@@ -0,0 +1,57 @@
+		icon_state = "raven_dead"
+	else
+		icon_state = "raven"
+	return
+/mob/living/simple_animal/pet/raven/verb/caw()
```
```suggestion
	return

/mob/living/simple_animal/pet/raven/verb/caw()
```
spacing

Sirryan2002 replied:
```suggestion
	
/mob/living/simple_animal/pet/raven/verb/caw()
```
spacing and hanging return (it's redundant since procs automatically return at the end of its definition)

Sirryan2002 commented on code/modules/mob/living/simple_animal/friendly/raven.dm:
```diff
@@ -0,0 +1,57 @@
+	set name = "caw"
+	set category = "Raven"
+	set desc = "caws"
+	if(cooldown < world.time - 20) // A cooldown, to stop people being jerks
+		playsound(src.loc, 'sound/creatures/caw.ogg', 50, 1)
+		cooldown = world.time
```
```suggestion
	if(cooldown < world.time) // A cooldown, to stop people being jerks
		playsound(src.loc, 'sound/creatures/caw.ogg', 50, 1)
		cooldown = world.time + 2 SECONDS
```
I'd prefer the cooldown be handled like this

Sirryan2002 commented on code/modules/mob/living/simple_animal/friendly/raven.dm:
```diff
@@ -0,0 +1,57 @@
+	set desc = "Sit on a nice comfy perch."
+
+	if(icon_state== "raven")
+		icon_state = "raven_dead"
```
this needs a different icon other than its dead icon

Raven567 replied:
It certainly does ha ha! yeah, unfortunately the animations take a little bit.

Raven567 commented:
I believe the project is in a pretty complete state! The pixel art is looking pretty decent to me, I am sure there are improvements always that could be made! The raven has two special abilities, The power to make an audible caw set on a cool down, also the ability to fly, and land. They can be found on  here. 
![image](https://user-images.githubusercontent.com/28615254/196017510-2d77c630-5559-47c6-a027-34d85f89c4bd.png)


Also id like to thank Sirryan2002 for helping me on every step of the journey! MrRamon and Azlan for your guys invaluable pixel art advise, and criticism, as well as Teri the community manager for emotional support. And I am probably not done, but this for my first commit has been a fantastic journey, just incase this is my last update.

Sirryan2002 commented on code/modules/mob/living/simple_animal/friendly/raven.dm:
```diff
@@ -0,0 +1,58 @@
+/mob/living/simple_animal/pet/raven//the perfectly normal raven.
```
```suggestion
///the perfectly normal raven.
/mob/living/simple_animal/pet/raven
```

Raven567 replied:
Merge issues...

Sirryan2002 commented on code/modules/mob/living/simple_animal/friendly/raven.dm:
```diff
@@ -0,0 +1,58 @@
+	can_collar = TRUE
+	gold_core_spawnable = FRIENDLY_SPAWN
+
+/mob/living/simple_animal/pet/raven/nevermore//a unique pet, I consider maybe the  icon states are are redundant?
```
```suggestion
///a unique raven pet
/mob/living/simple_animal/pet/raven/nevermore
```

Sirryan2002 commented on code/modules/mob/living/simple_animal/friendly/raven.dm:
```diff
@@ -0,0 +1,58 @@
+/mob/living/simple_animal/pet/raven/nevermore//a unique pet, I consider maybe the  icon states are are redundant?
+	name = "Nevermore"
+	desc = "The coroner's pet. Reminds us that peace and quiet is coming for us all."
+	icon_state = "raven"
+	icon_living = "raven"
+	icon_dead = "raven_dead"
```
```suggestion
```
all of these are redundant due to parent type

Sirryan2002 commented on code/modules/mob/living/simple_animal/friendly/raven.dm:
```diff
@@ -0,0 +1,58 @@
+	set category = "Raven"
+	set desc = "Sit on a nice comfy perch."
+
+	if(icon_state == "raven")//Changes the state to the flying animation if the check returns true.
```
again this should use another var called `perched` or something instead of relying on checking the icon state.

Sirryan2002 commented on code/modules/mob/living/simple_animal/friendly/raven.dm:
```diff
@@ -0,0 +1,58 @@
+	set category = "Raven"
+	set desc = "caws"
+	if(wait < world.time) // A wait, to stop people being jerks
+		playsound(src.loc, 'sound/creatures/caw.ogg', 50, 1)
```
```suggestion
		playsound(loc, 'sound/creatures/caw.ogg', 50, 1)
```
since this verb is defined on the raven object, you don't need any special key/object to access its methods and variables so `src` is basically "implied"

Sirryan2002 commented on code/modules/mob/mob.dm:
```diff
@@ -1538,3 +1538,4 @@ GLOBAL_LIST_INIT(holy_areas, typecacheof(list(
 	. = stat
 	stat = new_stat
 	SEND_SIGNAL(src, COMSIG_MOB_STATCHANGE, new_stat, .)
+var/wait = world.time //This varible allows you to set a cooldown on an ability, for an example usage reference the raven.dm, last line.
```
please scope this to 
`/mob/living/simple_animal/pet/raven` and name it something like `caw_cooldown` or whatnot. This is currently scoped **globally** but will only ever been used for the raven at the moment.

Raven567 replied:
Localized in the most recent commit!

Sirryan2002 commented on code/modules/mob/living/simple_animal/friendly/raven.dm:
```diff
@@ -0,0 +1,58 @@
+	else
+		icon_state = "raven"
+
+	return
```
again this return is redundant since procs automatically return at the end of execution
```suggestion
```

S34NW reviewed (changes requested):
You are likely going to have to redo mapping changes from scratch. I'm not sure what program you used to make the changes but it changed a whole load of other things within the map file for no reason. Please get the master maps and re-apply your changes over them using either the DM map editor, or SDMM2

Sirryan2002 reviewed (changes requested):
please see previous requested changes

Raven567 commented:
> You are likely going to have to redo mapping changes from scratch. I'm not sure what program you used to make the changes but it changed a whole load of other things within the map file for no reason. Please get the master maps and re-apply your changes over them using either the DM map editor, or SDMM2

I didn't change hardly anything, all it does is adds a single raven to each station.

Raven567 commented:
Ravens now spawn in the NT office in each station, also added them to the syndicate station. Completely started fresh with the maps so hopefully there are no more issues.

Henri215 commented on code/modules/mob/living/simple_animal/friendly/raven.dm:
```diff
@@ -0,0 +1,59 @@
+	maxHealth = 7
+	attacktext = "bites"
+	melee_damage_lower = 2
+	melee_damage_upper = 2
```
```suggestion
	melee_damage_upper = 2
	turns_per_move = 5
```
While testing the raven did move way too often, setting the turns_per_move to 5 (same as parrots) looks like the best option.

Miraviel commented:
Having more animals is cool, but making it spawn in Chapel, let alone the NT office makes no sense. I understand you wish to have a pet raven in game, but that doesn't mean everyone should deal with it every round.

We recently added bunnies because a few players wanted to have them, they are currently only accessible via supply pack crates. I suggest you do the same with ravens and remove them from the maps:

In `code\datums\supplypacks.dm`, add a raven crate below `/datum/supply_packs/organic/bunny` (copy-paste the bunny one and replace it with the raven's data, should be the same price), then in `code\game\objects\structures\crates_lockers\crittercrate.dm`, do the same at the end of the file. Then launch a test server, go to cargo, order one and see it works.

hal9000PR reviewed (changes requested):
Couple of things. Please don't wordlessly resolve reviews from people. Especially from the review team. Actually use their feedback. people put a lot of effort into reviewing code and its not just for you to ignore.

Second thing is, I'm not sure we need ravens mapped into 4 different maps. I think a supply crate be better.

Raven567 commented:
Closed due to some very strange issues I have been unable to resolve, I will be reopening a new pull in an hour or so.
